packages/plugin-ext/src/main/browser/custom-editors/custom-editor-opener.tsx

```diff
@@ -68,14 +69,21 @@ export class CustomEditorOpener implements OpenHandler {
         if (widget?.isAttached) {
             return this.shell.activateWidget(widget.id);
         }
+        const stringUri = uri.toString();
+        if (!widget && this.pendingGetWidgets.has(stringUri)) {
+            return;
+        }
         if (!widget) {
             const id = v4();
+            this.pendingGetWidgets.set(stringUri, Disposable.create(() => {
+                this.pendingGetWidgets.delete(stringUri);
+            }));
```

Comment:
I'd argue creating a `Disposable` in here is a bit over-engineered. Would a `() => void` suffice in this place?